### PR TITLE
Don't display spinners when stdout is not a terminal

### PR DESCRIPTION
**Pull request by petermattis**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/roachprod/45)

### CODE REVIEW

**Comment by petermattis** [comment]
The code to do this is a bit messy, but I don't want to invest energy in a more holistic approach at this time.

File: writer.go [hunk 1]
```diff
@@ -4,9 +4,14 @@ import (
 	"bytes"
 	"fmt"
 	"io"
+	"os"
 	"strings"
+
+	isatty "github.com/mattn/go-isatty"
```
Comment (benesch) [inline writer.go]: You could avoid this dep with https://godoc.org/golang.org/x/crypto/ssh/terminal#IsTerminal, I think.

Comment (petermattis) [reply writer.go]: Ok, let me try that.

**Comment by petermattis** [comment]
> In the future, could you separate out the dep changes into a separate commit, plz? It was hard to navigate the review, even when using github instead of reviewable and with most of them unexpanded

Yeah, I realize the dep changes obfuscated the actual code changes. If the dep changes are in a 2nd commit, then the first commit won't build. How do you put the dep changes in a first commit when the dependency doesn't exist yet?

File: writer.go (hunk 1 above)
Comment (petermattis) [reply writer.go]: Ok, that avoided the dependency on `mattn/go-isatty`, but `dep` still needed to pull in a bunch of new dependencies.

**Comment by petermattis** [comment]
@danhhz Ok, I'll try both your suggestions next time this comes up.

**Pull request merged**